cmd/access/node_builder/access_node_builder.go

```diff
@@ -590,14 +626,120 @@ func (builder *FlowAccessNodeBuilder) BuildExecutionSyncComponents() *FlowAccess
 				builder.State,
 				builder.Storage.Headers,
 				builder.executionDataConfig,
+				execDataDistributor,
 			)
 
 			builder.FollowerDistributor.AddOnBlockFinalizedConsumer(builder.ExecutionDataRequester.OnBlockFinalized)
-			builder.ExecutionDataRequester.AddOnExecutionDataReceivedConsumer(execDataDistributor.OnExecutionDataReceived)
+
+			// add requester into ReadyDoneAware dependency passed to indexer. This allows the indexer
+			// to wait for the requester to be ready before starting.
+			requesterDependable.Init(builder.ExecutionDataRequester)
 
 			return builder.ExecutionDataRequester, nil
 		})
 
+	if builder.executionDataIndexingEnabled {
+		var indexedBlockHeight storage.ConsumerProgress
+
+		builder.
+			Module("indexed block height consumer progress", func(node *cmd.NodeConfig) error {
+				// Note: progress is stored in the MAIN db since that is where indexed execution data is stored.
+				indexedBlockHeight = bstorage.NewConsumerProgress(builder.DB, module.ConsumeProgressExecutionDataIndexerBlockHeight)
+				return nil
+			}).
+			Module("events storage", func(node *cmd.NodeConfig) error {
+				builder.Storage.Events = bstorage.NewEvents(node.Metrics.Cache, node.DB)
+				return nil
+			}).
+			Module("transaction results storage", func(node *cmd.NodeConfig) error {
+				builder.Storage.LightTransactionResults = bstorage.NewLightTransactionResults(node.Metrics.Cache, node.DB, bstorage.DefaultCacheSize)
+				return nil
+			}).
+			DependableComponent("execution data indexer", func(node *cmd.NodeConfig) (module.ReadyDoneAware, error) {
+				// Note: using a DependableComponent here to ensure that the indexer does not block
+				// other components from starting while bootstrapping the register db since it may
+				// take hours to complete.
+
+				pdb, err := pStorage.OpenRegisterPebbleDB(builder.registersDBPath)
```

Comment:
Home comes this is not created as a module?

Reply:
it could be, but we only use it here so I figured it wasn't necessary.